compiler/rustc_hir_analysis/src/coherence/builtin.rs

```diff
@@ -386,8 +385,7 @@ pub fn coerce_unsized_info<'tcx>(tcx: TyCtxt<'tcx>, impl_did: DefId) -> CoerceUn
         (mt_a.ty, mt_b.ty, unsize_trait, None)
     };
     let (source, target, trait_def_id, kind) = match (source.kind(), target.kind()) {
-        (&ty::Ref(r_a, ty_a, mutbl_a), &ty::Ref(r_b, ty_b, mutbl_b)) => {
-            infcx.sub_regions(infer::RelateObjectBound(span), r_b, r_a);
```

Comment:
Why is this line removed? This has side-effects.

Reply:
You could give it a different subregion origin, but this is required for soundness :sweat:

Reply:
This code reproduces this error:

```rust
#![feature(coerce_unsized)]
#![feature(unsize)]

use core::marker::Unsize;
use core::ops::CoerceUnsized;

struct Wrapper<T>(T);

impl<'a, 'b, T, S> CoerceUnsized<&'a Wrapper<T>> for &'b Wrapper<S> where S: Unsize<T> {}
```

Reply:
Ahh, my mistake, I assumed that this error code was no longer used. I'll close this PR and document/test it instead.

Reply:
It's alright -- in general, I would *never* remove logic just because you think it's dead code. Sometimes people just forget to add a UI test for an error code.

Replace it with a `delay_span_bug` or `bug!()` if necessary, but unless you can statically prove that something is unreachable, you're very likely to introduce unsoundness or other subtle bugs into the compiler.

Reply:
Hmm, I was kind of naive. I assumed that untested error codes (basically all of them are like this) were from old Rust versions and had never been formally removed.